packages/server/src/core/bank_rule/specialization_groups.rs

```diff
@@ -47,19 +47,41 @@ fn is_valid_assignment(
     true
 }
 
+fn get_best_match(
```

Comment:
We don't check here that the user completed the mandatory courses.
Which means, if the user completed X courses from some sg, and we didn't find a valid assignment, we will mark this group as done.